osconfig_tests/test_suites/package_management/package_management.go

```diff
@@ -112,44 +108,197 @@ func TestSuite(ctx context.Context, tswg *sync.WaitGroup, testSuites chan *junit
 	logger.Printf("Finished TestSuite %q", testSuite.Name)
 }
 
-func runCreateOsConfigTest(ctx context.Context, testCase *junitxml.TestCase, logger *log.Logger) {
+func runCreateOsConfigTest(ctx context.Context, testCase *junitxml.TestCase, testSetup *packageManagementTestSetup, logger *log.Logger) {
 
-	osConfig := &osconfigpb.OsConfig{
-		Name: "create-osconfig-test",
+	oc := &osconfigserver.OsConfig{
+		&osconfigpb.OsConfig{
+			Name: "createosconfig-test-osconfig",
+		},
 	}
 
-	defer osconfigserver.CleanupOsConfig(ctx, testCase, logger, osConfig)
+	logger.Printf("create osconfig request:\n%s\n\n", dump.Sprint(oc))
 
-	req := &osconfigpb.CreateOsConfigRequest{
-		Parent:   "projects/compute-image-test-pool-001",
-		OsConfig: osConfig,
-	}
+	parent := fmt.Sprintf("projects/%s", testProject)
+	res, err := osconfigserver.CreateOsConfig(ctx, logger, oc, parent)
 
-	logger.Printf("create osconfig request:\n%s\n\n", dump.Sprint(req))
+	defer cleanuposconfig(ctx, testCase, logger, oc)
 
-	res, err := osconfigserver.CreateOsConfig(ctx, logger, req)
 	if err != nil {
 		testCase.WriteFailure("error while creating osconfig:\n%s\n\n", err)
 	}
 
 	logger.Printf("CreateOsConfig response:\n%s\n\n", dump.Sprint(res))
 }
 
+func runPackageInstallTest(ctx context.Context, testCase *junitxml.TestCase, testSetup *packageManagementTestSetup, logger *log.Logger) {
+
+	osConfig, err := osconfigserver.JsonToOsConfig(packageInstallTestOsConfigString, logger)
```

Comment:
Why are you specifying the configs as JSON strings instead of using plain structs with type safety?

Reply:
my initial idea was to have json data files for each osconfig, assignment, etc. this way the actual code does not become lengthy making it more readable.

Reply:
It's simpler and has better error handling just to have a go file with the various structs you want to use, then you can at least have comments.

Reply:
actually yeah, it was annoying at times to fix the json string. i will replace the json strings to structs and still keep all of them in a separate data file.

Reply:
moved away from creating the objects using json to creating them through structs.